src/main/java/MikeException.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * 
+ */
+package main.java;
+
+/**
+ * @author Mike
+ *
+ */
+public class MikeException extends Exception {
```

Comment:
The question asked for _exceptions_, i.e. a type for each case. `MikeException` doesn't really indicate what went wrong to the caller (unless you're manually computing the results behind the scenes).